Using markdown it to render

acrowfliedover opened the pull request:
-highlight buttons not yet working

enochzhouyz committed:
fix: open markdown
put markdown into the chat messages

acrowfliedover committed:
fix(markdown) add to config.js

acrowfliedover committed:
fix hovered question

acrowfliedover committed:
changing reader submodule

enochzhouyz committed:
fix: correct highlight button replacement
Find a workable solution with cursor to replace source tag text in place with source button

acrowfliedover committed:
change to using markdown-it

acrowfliedover committed:
add markdown it

acrowfliedover committed:
add latex display

acrowfliedover committed:
add more fonts doesn't help yet

acrowfliedover committed:
markdown display alright

acrowfliedover committed:
table display works

acrowfliedover committed:
add back handle source click function

enochzhouyz committed:
Merge branch 'staging' into leo-markdown-it

enochzhouyz commented:
Update: I merged the conflict at my judgement. After merging, L1673 has a syntax error of using "continue" without for loop, which might be the consequence of merge conflict.

enochzhouyz commented:
The usage on WSL Linux is not successful. The error is as follows: 
![Screenshot 2025-06-26 184035](https://github.com/user-attachments/assets/8c4bb8a1-acf1-4891-9de6-63c37765173f)

enochzhouyz commented:
![Screenshot 2025-06-26 184627](https://github.com/user-attachments/assets/f7f91431-133a-46de-b031-9b31ddc3a230)

enochzhouyz commented:
I recommend not to merge this pr. It might be a good idea that I try to implement the streaming components (which might resolve above error) before merging pr

enochzhouyz committed:
fix: minor conflicts

enochzhouyz committed:
fix: add streaming component
used cursor to complete three streaming component files and completed chat process in chat box file, added streaming process and try to fix some issue in parsing tags

enochzhouyz commented:
Updates: 
1. Fix minor bugs after resolving merge conflict
2. Attempted to fix bug in processing some tags like (approximately) <br/g 
3. Used cursor to complete three streaming component files and completed chat process in chat box file, added streaming process and try to fix some issue in parsing tags
4. Currently, there is problem in some session, but the chat works fine for some other sessions. I recommend that the team keep testing and continue to fix the code based on current progress.

enochzhouyz committed:
fix: add highlight button
try to add highlight button to text

enochzhouyz committed:
fix: highlight source button click
source button click is now effective

enochzhouyz committed:
fix: table with highlight
used cursor to use a different approach to render source button by runtime replacement, allowing table to deploy well with source buttons

enochzhouyz committed:
fix: random design fix
make following attempts to fix the XML parsing issue, including react-html-parsing package, other packages, refactor regex conversion, remove runtime source replacement, add adjustment for math symbols, etc

enochzhouyz committed:
fix: possible solution to xml parsing
find a possible solution by using alternative parsing tools like xmldom parsing, etc., with proper preprocessing of html5 text

enochzhouyz committed:
fix: uncomment code
uncomment code for math expressions and source button

enochzhouyz committed:
Merge branch 'staging' into leo-markdown-it
